crates/blockchain/mempool.rs

```diff
@@ -100,8 +94,8 @@ pub fn filter_transactions(
 }
 
 /// Remove a transaction from the mempool
-pub fn remove_transaction(hash: H256, store: &Store) -> Result<(), StoreError> {
-    store.remove_transaction_from_pool(hash)
+pub fn remove_transaction(address: Address, nonce: u64, store: &Store) {
```

Comment:
why are you indexing with address + nonce? Technically there could be more than 1 tx with the same address and nonce.

Reply:
`address` here means sender address, I'll change that to make it more clear. 

I don't believe there can be more than 1 tx with same sender and nonce, if someone sends a second tx with the same nonce we either reject it if the fee is lower or replace the current one with it if the fee is higher

Reply:
This has been updated, we now index with `sender_address` and `tx_hash`.